### PR TITLE
[GVN] Handle empty attrs in Expression ==

### PR DESCRIPTION
**Pull request by macurtis-amd**

### CODE REVIEW

**Comment by dtcxzyw** [comment]
Does it address the crash reported in https://github.com/llvm/llvm-project/pull/114545#issuecomment-2465910622?
cc @ronlieb

**Comment by ronlieb** [comment]
> Does it address the crash reported in [#114545 (comment)](https://github.com/llvm/llvm-project/pull/114545#issuecomment-2465910622)? cc @ronlieb

yes it does, thanks!

**Review by ronlieb — APPROVED** [review]

**Review by dtcxzyw — APPROVED** [review]
LG

File: llvm/test/Transforms/GVN/intersect-empty-attr.ll [hunk 1]
```diff
@@ -0,0 +1,19 @@
+; RUN: opt -S -passes=gvn < %s | FileCheck %s
```
Comment (nikic) [inline llvm/test/Transforms/GVN/intersect-empty-attr.ll:1]: Use update_test_checks.py please.

Comment (macurtis-amd) [reply llvm/test/Transforms/GVN/intersect-empty-attr.ll:2]: Done

**Review by nikic — APPROVED** [review]
LGTM

**Pull request merged**